libraries/rush-lib/src/logic/pnpm/PnpmProjectManifestConfiguration.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { JsonFile, JsonSchema, FileSystem } from '@rushstack/node-core-library';
+
+import type { ProjectManifest } from '@pnpm/types';
+
+/**
+ * Pnpm config in package.json
+ * @beta
+ */
+export type IPnpmProjectManifestConfigurationJson = Required<ProjectManifest>['pnpm'];
```

Comment:
I'd honestly consider just setting this type to `unknown`. It's none of Rush's business what is actually in there, validation is entirely on `pnpm`.